controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -311,116 +311,74 @@ bool master_thread::handle_cmdu_1905_1_message(const std::string &src_mac,
-            tlvAutoconfigFreqBand = cmdu_rx.addClass<ieee1905_1::tlvAutoconfigFreqBand>();
-            break;
-        case int(wfa_map::eTlvTypeMap::TLV_SEARCHED_SERVICE):
-            LOG(DEBUG) << "Found TLV_SEARCHED_SERVICE TLV";
-            tlvSearchedService = cmdu_rx.addClass<wfa_map::tlvSearchedService>();
-            break;
-        default:
-            LOG(ERROR) << "Unknown TLV, type " << std::hex << type << ", dropping...";
-            return false;
-            break;
-        }
-    }
-
-    if (tlvAlMacAddressType) {
-        al_mac =
-            network_utils::mac_to_string((const unsigned char *)tlvAlMacAddressType->mac().oct);
-        LOG(DEBUG) << "mac=" << al_mac;
-    } else {
-        LOG(ERROR) << "tlvAlMacAddressType missing - ignoring autconfig search message";
+    auto tlvAlMacAddressType = cmdu_rx.getClass<ieee1905_1::tlvAlMacAddressType>();
+    if (!tlvAlMacAddressType) {
+        LOG(ERROR) << "getClass<tlvAlMacAddressType> failed";
         return false;
     }
-
-    if (tlvSearchedRole) {
-        LOG(DEBUG) << "searched_role=" << int(tlvSearchedRole->value());
-        if (tlvSearchedRole->value() != ieee1905_1::tlvSearchedRole::REGISTRAR) {
-            LOG(ERROR) << "invalid tlvSearchedRole value";
-            return false;
-        }
-    } else {
-        LOG(ERROR) << "tlvSearchedRole missing - ignoring autconfig search message";
+    auto tlvSearchedRole = cmdu_rx.getClass<ieee1905_1::tlvSearchedRole>();
+    if (!tlvSearchedRole) {
+        LOG(ERROR) << "getClass<tlvSearchedRole> failed";
         return false;
     }
-
+    auto tlvAutoconfigFreqBand = cmdu_rx.getClass<ieee1905_1::tlvAutoconfigFreqBand>();
     if (!tlvAutoconfigFreqBand) {
-        LOG(ERROR) << "addClass ieee1905_1::tlvAutoconfigFreqBand failed";
+        LOG(ERROR) << "getClass<tlvAutoconfigFreqBand> failed";
+        return false;
+    }
+    auto tlvSupportedServiceIn = cmdu_rx.getClass<wfa_map::tlvSupportedService>();
+    if (!tlvSupportedServiceIn) {
+        LOG(ERROR) << "getClass<tlvSupportedService> failed";
+        return false;
+    }
+    auto tlvSearchedService = cmdu_rx.getClass<wfa_map::tlvSearchedService>();
+    if (!tlvSearchedService) {
+        LOG(ERROR) << "getClass<tlvSearchedService> failed";
+        return false;
     }
 
-    auto &auto_config_freq_band = tlvAutoconfigFreqBand->value();
-    if (tlvAutoconfigFreqBand) {
-        LOG(DEBUG) << "band=" << int(auto_config_freq_band);
-    } else {
-        LOG(ERROR) << "tlvAutoconfigFreqBand missing - ignoring autconfig search message";
+    auto al_mac =
+        network_utils::mac_to_string((const unsigned char *)tlvAlMacAddressType->mac().oct);
+    LOG(DEBUG) << "mac=" << al_mac;
+
+    LOG(DEBUG) << "searched_role=" << int(tlvSearchedRole->value());
+    if (tlvSearchedRole->value() != ieee1905_1::tlvSearchedRole::REGISTRAR) {
+        LOG(ERROR) << "invalid tlvSearchedRole value";
```

Comment:
Continuing my previous comment, this is probably a warning as well.
Also, a good idea would be to print the received value... Will help debugging.

Reply:
Same anwer - perhaps you're right, but I didn't change the error logging as part of this PR and I don't want to do that either.